storm-client/src/jvm/org/apache/storm/messaging/netty/Server.java

```diff
@@ -260,7 +260,15 @@ public void channelActive(Channel c) {
 
     @Override
     public void received(Object message, String remote, Channel channel) throws InterruptedException {
-        List<TaskMessage> msgs = (List<TaskMessage>) message;
+        List<TaskMessage> msgs;
+
+        try {
+            msgs = (List<TaskMessage>) message;
+        } catch (ClassCastException e) {
+            LOG.error("Worker netty server receive message other than the expected class List<TaskMessage> from remote: {}", remote, e);
```

Comment:
nit: `receive` -> `received` ;

And ad  the log to indicate the message is ignored will be helpful